subsys/mpsl/mpsl_init.c

```diff
@@ -155,7 +155,10 @@ static int mpsl_lib_init(const struct device *dev)
 	clock_cfg.skip_wait_lfclk_started =
 		IS_ENABLED(CONFIG_SYSTEM_CLOCK_NO_WAIT);
 
-#ifdef CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION
+#ifdef CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC
+#ifndef CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION
+#error "MPSL requires clock calibration to be enabled when RC is used as LFCLK"
```

Comment:
Couldn't this be expressed in Kconfig? If not a BUILD_ASSERT would be better I think.

Reply:
I'm not sure how to express it in Kconfig in a nice way. `CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION` has a dependency to `CLOCK_CONTROL_NRF_K32SRC_RC`. Therefore we can't make `CLOCK_CONTROL_NRF_K32SRC_RC` select `CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION`. As `CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION` is a bool, we can't change its range. 

I can change it to use a build assert if you prefer that.

Reply:
I would have thought this:
```
	# MPSL requires clock calibration to be enabled when RC is used as LFCLK
	select CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION if CLOCK_CONTROL_NRF_K32SRC_RC
```
But looks like it doesn't work exactly how I expected it too.

Let's change it to a BUILD_ASSERT then.

Reply:
Tried using `BUILD_ASSERT(CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION, ""`. That does not work as `CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION` is not defined when set to `n`.